server/aesdsocket.c

```diff
@@ -0,0 +1,326 @@
+// Tester code for assignment 5 part 1
+// File:	writer.c
+// Author:	Mukta Darekar
+// Reference: https://docs.oracle.com/cd/E19620-01/805-4041/sockets-47146/index.html
+
+
+// Pre-processor directives
+#include <stdlib.h>
+#include <stdint.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <netinet/in.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include <string.h>
+#include <ctype.h>
+#include <syslog.h>
+#include <unistd.h>
+#include <arpa/inet.h>
+#include <signal.h>
+
+// Macro directives
+#define TOTAL_MSG	3	
+#define ERROR		1
+#define SUCCESS		0
+#define MYPORT		9000
+#define DEF_FILEPATH	"/var/tmp/aesdsocketdata"
+
+int sockfd = 0;
+char *sendbuffer = NULL;
+int acceptedfd = 0;
+int fd = 0;
+sigset_t mask;
+
+//filename and path in one variable
+const char *filepath = DEF_FILEPATH;
+
+//Function:	static void signal_handler(int signo)
+//Inputs:	signo - Signal number
+
+static void closeall(void)
+{
+	if(sendbuffer != NULL)	
+		free(sendbuffer);
+
+	if(acceptedfd)
+		close(acceptedfd);
+
+	if(sockfd)
+		close(sockfd);
+
+	if(fd)	
+	{
+		close(fd);
+		remove(filepath);
+	}
+	
+	closelog();	
+}
+
+//Function:	static void signal_handler(int signo)
+//Inputs:	signo - Signal number
+
+static void signal_handler(int signo)
+{
+
+	syslog(LOG_DEBUG, "in handler\n");
+	
+	if(signo == SIGINT)
+		syslog(LOG_DEBUG, "Caught signal SIGINT, exiting\n");
+	else
+		syslog(LOG_DEBUG, "Caught signal SIGTERM, exiting\n");
+
+	closeall();
+
+	exit(EXIT_SUCCESS);
```

Comment:
@MuktaDarekar you should avoid exiting in your signal handler

Reply:
done